crates/uv-configuration/src/vcs_options.rs

```diff
@@ -0,0 +1,123 @@
+use std::path::{Path, PathBuf};
+use std::process::{Command, Stdio};
+use std::str::FromStr;
+
+use anyhow::{Context, Result};
+use serde::Deserialize;
+
+use uv_fs::Simplified;
+use uv_warnings::warn_user;
+
+/// The version control system to use.
+#[derive(Clone, Copy, Debug, PartialEq, Default, Deserialize)]
+#[serde(deny_unknown_fields, rename_all = "kebab-case")]
+#[cfg_attr(feature = "clap", derive(clap::ValueEnum))]
+#[cfg_attr(feature = "schemars", derive(schemars::JsonSchema))]
+pub enum VersionControl {
+    /// Use Git for version control.
+    #[default]
+    Git,
+
+    /// Do not use version control.
+    None,
+}
+
+impl VersionControl {
+    /// Initializes the VCS system based on the provided path.
+    pub fn init(&self, path: &PathBuf) -> Result<()> {
+        match self {
+            VersionControl::None => Ok(()),
+            VersionControl::Git => Self::init_git(path),
+        }
+    }
+
+    fn init_git(path: &PathBuf) -> Result<()> {
+        let Ok(git) = which::which("git") else {
+            anyhow::bail!("could not find `git` in PATH");
+        };
```

Comment:
Does this mean if you don't have `git` installed and don't pass `--vcs git` we'll fail instead of just not using `git`? We might want to avoid that, either by returning a specific error variant here that the caller can ignore or by creating a separate case for the "default" behavior?

Reply:
I see this is handled downstream in a way quite similar to my note :) https://github.com/astral-sh/uv/pull/5476/files#r1762250146